pennylane_lightning/lightning_kokkos/lightning_kokkos.py

```diff
@@ -834,3 +835,32 @@ def processing_fn(tape):
                 return self.adjoint_jacobian(new_tape, starting_state, use_device_state)
 
             return processing_fn
+
+    @staticmethod
+    def get_c_interface():
+        """Returns a tuple consisting of the device name, and
+        the location to the shared object with the C/C++ device implementation.
+        """
+
+        lib_name = "liblightning_kokkos_catalyst.so"
+        package_root = os.path.dirname(__file__)
+
+        # The absolute path of the plugin shared object varies according to the installation mode.
+
+        # Wheel mode:
+        # Fixed location at the root of the project
+        wheel_mode_location = os.path.join(package_root, "..", lib_name)
+        if os.path.isfile(wheel_mode_location):
+            return "LightningKokkosSimulator", wheel_mode_location
+
+        # Editable mode:
+        # The build directory contains a folder which varies according to the platform:
+        #   lib.<system>-<architecture>-<python-id>"
+        # To avoid mismatching the folder name, we search for the shared object instead.
+        # TODO: locate where the naming convention of the folder is decided and replicate it here.
+        editable_mode_path = os.path.join(package_root, "../../build")
```

Comment:
Thanks for taking care of editable mode!
Could you check the support on macOS?

Reply:
Confirmed: it is located under `/Users/<user_name>/<path_to_pennylane-lightning_src>/build/lib.macosx-14.0-x86_64-cpython-312/pennylane_lightning` in my test machine, meaning the build folder is the same for all platforms